<launch>

  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="model" default="$(find mech_ros)/robot_description/urdf/mechROS.xacro"/>
<!--<arg name="model" default="/home/patrik/catkin_ws/src/mech_ros/robot_description/urdf/mechROS.urdf.xacro"/>-->
  <arg name="x_pos" default="-3.5"/>
  <arg name="y_pos" default="-2.5"/>
  <arg name="z_pos" default="0.03"/>
  
 <!-- Marker detection -->
 <!-- <arg name="markerSize"      default="0.1778"/>
 <arg name="ref_frame"       default="/mechROS_base_link"/> -->

   <!-- <remap from="/pi_camera/camera_info" to="/camera_info" />
   <remap from="/pi_camera/image_raw" to="/image" /> -->

	<!-- <arg name="aruco_eye_ros_node_name" 	default="aruco_eye_ros_node" /> -->

  <!-- <arg name="aruco_list_file" default="$(find aruco_eye_ros)/config/arUcoList_modified.xml"/> -->

  <!-- <arg name="camera_calibration_file" default="$(find mech_ros)/Config_ARuco/camera_Gazebo2.yml"/> -->
  <!-- <arg name="camera_calibration_file" default=""/> -->
  <!-- <arg name="camera_info_topic_name" default="/camera_info"/> -->

  <!-- <arg name="aruco_detector_frame_name" default="/front_camera_link"/> -->
  <!-- <arg name="aruco_marker_child_base_name" default="aruco_marker_"/> -->

  <!-- <arg name="image_topic_name" 	default="/image" /> -->
  <!-- <arg name="aruco_list_topic_name" 	default="/markers" /> -->

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
<!--    <arg name="debug" value="$(arg debug)" />-->
    <arg name="world_name" value="$(find mech_ros)/Worlds/Muzeum_AR_upraveno4.world"/>
    <!-- <arg name="world_name" value="$(find mech_ros)/Worlds/Blank_world.world"/> -->
    <arg name="verbose" value="true"/>
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />

  <!-- push robot_description to factory and spawn robot in gazebo -->
<!--  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"-->
<!--        args="-z 0.2 -unpause -urdf -model robot -param robot_description" respawn="false" output="screen" />-->

  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
        args="-x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -urdf -model robot -param robot_description" respawn="false" output="screen" />

  <node pkg="robot_state_publisher" type="robot_state_publisher"  name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node>

   <!-- <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher"> -->
    <!-- <param name="image_is_rectified" value="False"/> -->
    <!-- <param name="marker_size"        value="$(arg markerSize)"/> -->
    <!-- <param name="reference_frame"    value="$(arg ref_frame)"/> -->
    <!-- <param name="camera_frame"       value="/front_camera_link"/> -->
   <!-- </node> -->

<!-- Tohle bylo pouzito -->
	<!-- <node name="$(arg aruco_eye_ros_node_name)" pkg="aruco_eye_ros" type="aruco_eye_ros_detector" output="screen">

    <param name="aruco_list_file" value="$(arg aruco_list_file)" type="string"/>

    <param name="camera_calibration_file" value="$(arg camera_calibration_file)" type="string"/>
    <param name="camera_info_topic_name" value="$(arg camera_info_topic_name)" type="string"/>

    <param name="aruco_detector_frame_name" value="$(arg aruco_detector_frame_name)" type="string"/>
    <param name="aruco_marker_child_base_name" value="$(arg aruco_marker_child_base_name)" type="string"/>

    <param name="image_topic_name" value="$(arg image_topic_name)" type="string"/>
    <param name="aruco_list_topic_name" value="$(arg aruco_list_topic_name)" type="string"/>

	</node> -->

<!-- Nejstarsi verze -->
  <!-- <node name="aruco_eye_ros_display_node" pkg="aruco_eye_ros" type="aruco_eye_ros_display" output="screen"> -->

    <!-- <param name="camera_calibration_file" value="$(arg camera_calibration_file)" type="string"/> -->
    <!-- <param name="camera_info_topic_name" value="$(arg camera_info_topic_name)" type="string"/> -->

    <!-- <param name="display_output_image_window_name" value="aruco_image" type="string"/>     -->
    <!-- <param name="flag_display_output_image" value="False" type="Boolean"/> -->

    <!-- <param name="display_output_image_service_name" value="arg_display_image" type="string"/> -->

    <!-- <param name="image_topic_name" value="$(arg image_topic_name)" type="string"/> -->
    <!-- <param name="output_image_topic_name" value="aruco_image" type="string"/> -->
    <!-- <param name="aruco_list_topic_name" value="$(arg aruco_list_topic_name)" type="string"/> -->

	<!-- </node> -->

</launch>
